plugin/src/main/java/org/opensearch/ml/rest/RestMLExecuteAction.java

```diff
@@ -130,7 +131,8 @@ private void sendResponse(RestChannel channel, MLExecuteTaskResponse response) t
     }
 
     private void reportError(final RestChannel channel, final Exception e, final RestStatus status) {
-        channel.sendResponse(new BytesRestResponse(status, ErrorMessageFactory.createErrorMessage(e, status.getStatus()).toString()));
+        ErrorMessage errorMessage = ErrorMessageFactory.createErrorMessage(e, status.getStatus());
+        channel.sendResponse(new BytesRestResponse(RestStatus.fromCode(errorMessage.getStatus()), errorMessage.toString()));
```

Comment:
possible to add header `"application/json; charset=UTF-8"` in the response here or convert error to a XContentBuilder instead of `errorMessage.toString()`?

Reply:
Sure, I will send another pr to change it.